pkg/splunk/controller/util.go

```diff
@@ -308,3 +311,20 @@ func MergeServiceSpecUpdates(ctx context.Context, current *corev1.ServiceSpec, r
 
 	return result
 }
+
+// compareProbeValues compares the current and revised probe values
+func compareProbeValues(currentProbe *corev1.Probe, revisedProbe *corev1.Probe) bool {
```

Comment:
nit: the name of the function does not match what it is doing. Maybe something like `hasProbesChanged` or something like that..?